frontend/src/components/S3Modal.tsx

```diff
@@ -16,17 +16,26 @@ const S3Modal: React.FC<S3ModalProps> = ({ hideModal, open }) => {
   const [statusMessage, setStatusMessage] = useState<string>('');
   const { userCredentials } = useCredentials();
   const { setFiles, setFilesData } = useFileContext();
+
   const changeHandler = (e: any) => {
     setBucketUrl(e.target.value);
   };
   const submitHandler = async (bucketUrl: string) => {
+    if (bucketUrl && bucketUrl[bucketUrl.length - 1] != '/') {
+      setBucketUrl((bucketUrl += '/'));
+    }
     if (accessKey.length) {
       localStorage.setItem('accesskey', accessKey);
```

Comment:
I wonder if it wouldn't be better/easier to maintain in the future if instead of having different items in localStorage, we have one object containing all
So instead of having'accesskey' , 'secretkey' , 'neo4jconnection' etc we only have one item : 'nameOfYourApp' which would be an object like nameOfYourApp: {s3AccessKey: 'xxx', s3SecretKey: 'yyy', neo4jConnection: {uri...}, etc...}
This is how we tend to usually do it in other Neo4j apps I believe

Reply:
sure, we will maintain in one object just for an ease getting we kept like that we will move to one single object in coming days.